tensorflow/lite/micro/tools/make/targets/stm32f4_makefile.inc

```diff
@@ -0,0 +1,86 @@
+# Settings for stm32f4 based platforms
+ifeq ($(TARGET), stm32f4)
+  export PATH := $(MAKEFILE_DIR)/downloads/gcc_embedded/bin/:$(PATH)
+  TARGET_ARCH := cortex-m4
+  TARGET_TOOLCHAIN_PREFIX := arm-none-eabi-
+
+  $(eval $(call add_third_party_download,$(GCC_EMBEDDED_URL),$(GCC_EMBEDDED_MD5),gcc_embedded,))
+  $(eval $(call add_third_party_download,$(CMSIS_URL),$(CMSIS_MD5),cmsis,))
+  $(eval $(call add_third_party_download,$(STM32_BARE_LIB_URL),$(STM32_BARE_LIB_MD5),stm32_bare_lib,))
+
+  PLATFORM_FLAGS = \
+    -DGEMMLOWP_ALLOW_SLOW_SCALAR_FALLBACK \
+    -DTF_LITE_STATIC_MEMORY \
+    -DTF_LITE_MCU_DEBUG_LOG \
+    -fno-rtti \
+    -fmessage-length=0 \
+    -fno-exceptions \
+    -fno-unwind-tables \
+    -fno-builtin \
+    -ffunction-sections \
+    -fdata-sections \
+    -funsigned-char \
+    -MMD \
+    -mcpu=cortex-m4 \
+    -mthumb \
+    -std=gnu++11 \
+    -Wvla \
+    -Wall \
+    -Wextra \
+    -Wno-unused-parameter \
+    -Wno-missing-field-initializers \
+    -Wno-write-strings \
+    -Wno-sign-compare \
+    -fno-delete-null-pointer-checks \
+    -fomit-frame-pointer \
+    -fpermissive \
+    -g \
+    -Os
+  CXXFLAGS += $(PLATFORM_FLAGS)
+  CCFLAGS += $(PLATFORM_FLAGS)
+  LDFLAGS += \
+    --specs=nosys.specs \
+    -T $(MAKEFILE_DIR)/targets/stm32f4/stm32f4.lds \
+    -Wl,-Map=$(MAKEFILE_DIR)/gen/$(TARGET).map,--cref \
+    -Wl,--gc-sections
+  BUILD_TYPE := micro
+  MICROLITE_LIBS := \
+    -lm
+  INCLUDES += \
+    -isystem$(MAKEFILE_DIR)/downloads/cmsis/CMSIS/Core/Include/ \
+    -I$(MAKEFILE_DIR)/downloads/stm32_bare_lib/include
+  MICROLITE_CC_SRCS += \
+    $(wildcard $(MAKEFILE_DIR)/downloads/stm32_bare_lib/source/*.c) \
+    $(wildcard $(MAKEFILE_DIR)/downloads/stm32_bare_lib/source/*.cc)
+  EXCLUDED_SRCS := \
+    $(MAKEFILE_DIR)/downloads/stm32_bare_lib/source/debug_log.c
+  MICROLITE_CC_SRCS := $(filter-out $(EXCLUDED_SRCS), $(MICROLITE_CC_SRCS))
+  # Stm32f4 is reusing the bluepill renode scripts for now
+  TEST_SCRIPT := tensorflow/lite/micro/testing/test_bluepill_binary.sh
```

Comment:
This PR looks great overall, but I'm confused about how the runtime testing with Renode works. I thought the test_bluepill_binary.sh script invoked Renode with a Cortex M3 target, but it looks like you're building for a Cortex M4? Can you help me understand what's going on here? Thanks!

Reply:
Well in the beginning we decided to introduce a new M4 target since we knew that was working on real HW on that particular branch/commit and we didn't know the last time when the Bluepill target was working. Then it turns out the two targets are quite similar from Renode point of view. Looking in ./tensorflow/lite/micro/testing/bluepill.resc, there is the line: machine LoadPlatformDescription @platforms/cpus/stm32f103.repl.
When I started testing, I used: machine LoadPlatformDescription @platforms/cpus/stm32f4.repl, which I would expect for M4, and that is about the only difference. However it turned out it doesn't really matter so I decided not to copy every Bluepill file, which would otherwise be required. Michael Gielda at Antmicro, answered: "Well, how those TF Lite Micro examples work, it probably does not really matter which CPU core they run on, but of course I can ask the team. Your cores are pretty compatible if you don't use the fancy stuff."

Reply:
After all we decided to copy new files from the bluepill files to get the correct platform description for M4.
We may use the "fancy stuff" as there a lot of optimizations in CMSIS, and it would also otherwise be confusing for a user.